app/main.py

```diff
@@ -1,25 +1,95 @@
 from abc import ABC
 
 
+class Visitor:
+    def __init__(
+            self,
+            name: str,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
 
+    def __set_name__(
+            self,
+            owner: type,
+            name: str
+    ) -> None:
+        self.protected_name = f"_{name}"
 
-class Visitor:
-   pass
+    def __get__(
+            self,
+            instance: Visitor,
+            owner: type
+    ) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: Visitor,
+            value: int
+    ) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value should not be "
+                             f"less than {self.min_amount} "
+                             f"and greater than {self.max_amount}")
+        setattr(instance, self.protected_name, value)
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: [SlideLimitationValidator]
```

Comment:
Indicate here that you expect to obtain not an instance of the `SlideLimitationValidator` class inside of a list, but a subclass of the `SlideLimitationValidator` class using the following type annotation:
```suggestion
            limitation_class: type[SlideLimitationValidator]
```

Reply:
done
thanks for comment